### PR TITLE
Add support for configuring zstdCompressorCallback

### PR DESCRIPTION
**Pull request by skarimo**
Add ability to configure `ZstdCompressorCallback` so users are able to supply their own zstd compressor callback method.  Example usage:

```
import { compressSync } from 'zstd.ts'

const conf = {
  zstdCompressorCallback: (body: string) => compressSync({input: Buffer.from(body, "utf8")})
}
const configuration = client.createConfiguration(conf);
const apiInstance = new v2.MetricsApi(configuration);
```

### CODE REVIEW

**Review by therve — COMMENTED** [review]
LGTM! Please document it in the README. Do we need to set the callback in tests?

File: .generator/src/generator/templates/configuration.j2 [hunk 1]
```diff
@@ -101,6 +105,7 @@ export function createConfiguration(conf: ConfigurationParameters = {}): Configu
     httpConfig: conf.httpConfig || {},
     debug: conf.debug
   };
+  configuration.httpApi.zstdCompressorCallback = conf.zstdCompressorCallback || undefined
```
Comment (NouemanKHAL) [inline .generator/src/generator/templates/configuration.j2:108]:
```suggestion
  configuration.httpApi.zstdCompressorCallback = conf.zstdCompressorCallback
```
I guess it'll be `undefined` by default if it wasn't set in the config, right?

File: README.md [hunk 2]
```diff
@@ -161,9 +161,54 @@ async function main() {
 main();
 ```
 
+### Zstd compression
+
+Zstd compression support requires users to supply their own zstd compressor callback function.
+The callback should accept string arg and return compressed Buffer data.
+Callback signature: (body: string) => Buffer
```
Comment (NouemanKHAL) [inline README.md:168]:
```suggestion
`Callback signature: (body: string) => Buffer`
```
Would probably be more readable this way

File: .generator/src/generator/templates/http/isomorphic-fetch.j2 [hunk 3]
```diff
@@ -24,6 +25,12 @@ export class IsomorphicFetchHttpLibrary implements HttpLibrary {
         body = bufferFrom(pako.gzip(body).buffer);
       } else if (headers["Content-Encoding"] == "deflate") {
         body = bufferFrom(pako.deflate(body).buffer);
+      } else if (headers["Content-Encoding"] == "zstd1") {
+        if (this.zstdCompressorCallback) {
+          body = bufferFrom((this.zstdCompressorCallback(body)).buffer);
```
Comment (NouemanKHAL) [inline .generator/src/generator/templates/http/isomorphic-fetch.j2:30]: Why do we use `bufferFrom` if the callback function returns a `Buffer` object?

**Review by NouemanKHAL — APPROVED** [review]
LGTM!

**Pull request merged**